### ISSUE TITLE
 [FEATURE REQUESTS]

### ISSUE THREAD

**Issue by stefffkec** [QuartzWarrior/Tip.cc-Autocollect#21]
1. Hello! I have a problem. I'm claiming redpackets (envelopes) without a problem, until someone comes to a very great idea of detecting scripters by doing "$redpacket 1 witchfire 1s" like 10 times in a row, which all get claimed by the script. Then I get banned from the server. Is there a way to bypass this, while still claiming some bigger envelopes? Suggestion: In the current settings, you can ignore anything under $x.xx, make it automatically ignore anything without value. This should only be for the envelopes. The script should still claim no-value currencies for airdrops, phrasedrops and anything else.

2. Is it possible to do multiaccounting? (using tokens.txt)

3. Following up on 2, the script could have a setting of switching between the envelopes. First envelope is claimed by the first account, second envelope by the second account etc. This would bypass people saying "QuartzWarrior claimed the last 10 envelopes, ban him!"
4. The script could have a randomized CPM. I used 2 accounts and for all phrasedrops they were always first and second to say the phrase, which got me banned on both accounts. So the user could set a minimum and maximum CPM, and the script will randomize it. 
5. Thanks for reading!

**Comment by QuartzWarrior** [comment]
1. I do have the feature 'Earnings Threshold: Checks if earnings are above a certain amount before claiming.' (`IGNORE_DROPS_UNDER`), but you could also use `IGNORE_THRESHOLDS`, which is in format `value:chance_in_percent` which has a chance to ignore any drop under that value.
2. Thats only in my premium golang version right now, I might add it in the future though.
3. Thats a valid suggestion, might add it to the premium version.
4. okay yeah, thats a good one. u could do a random delay, which could separate them a bit. Might implement this.

**Comment by stefffkec** [comment]
I don't have money to buy the premium version, but where can I find it?

Am Mo., 14. Okt. 2024 um 04:30 Uhr schrieb Isaiah ***@***.***

**Comment by stefffkec** [comment]
Also redpackets don't work anymore, idk why

**Comment by stefffkec** [comment]
Another suggestion: You have the "whitelist servers where the bot should work". But can you also do "whitelist channels only"? Cause people create a #botcheck channel and just create a $0.0000001 airdrop, which gets picked up by the bot

**Comment by QuartzWarrior** [comment]
> I don't have money to buy the premium version, but where can I find it?

https://shop.quartzwarrior.xyz/product/tipcc-go

> Also redpackets don't work anymore, idk why

ill look into it





> Another suggestion: You have the "whitelist servers where the bot should work". But can you also do "whitelist channels only"? Cause people create a #botcheck channel and just create a $0.0000001 airdrop, which gets picked up by the bot

alright, yeah, so add channel whitelist as well. alright.

**Comment by stefffkec** [comment]
I tested out a bit and realized that the script not claiming envelopes occurs when you have smart delay turned on. Hope that helps!

**Comment by QuartzWarrior** [comment]
yes, this is because envelopes are a 1 person claim. Smart delay waits a quarter of the time until the drop ends. so, if someone claims it before the wait time ends, it obviously won't claim it.

**Comment by stefffkec** [comment]
No no I meant it doesn't even try to claim it when there is no time limit. Maybe the smart delay could be only for airdrops/phrasedrops/mathdrops

**Comment by QuartzWarrior** [comment]
there is always a time limit... I could add a feature where you can customize where the smart delay is used tho

**Comment by stefffkec** [comment]
Thanks for the update! But there is an error. After the setup (on the first run), it's all good. On the second run (close and reopen the script), the script just opens and instantly closes (I do "py tipcc_autocollect.py" and another cmd just opens closes)

**Comment by stefffkec** [comment]
Nevermind, I am sorry. It only happened since I duplicated the folder to use 2 accounts at the same time. I'll just do the setup for the other account as well

**Comment by QuartzWarrior** [comment]
alr..

**Comment by jamestaylor141** [comment]
Hey, sorry for just "throwing" in some new suggestions, it's just as I use the script more and more, I get banned more often, and I find new ways to bypass that. Here they are, you can code the ones you like:
1. There is the feature "ignore drops under $x.xx". And that works fine. But are you able to make a feature "ignore all redpackets under $x.xx ?" If I set this to $0.5, then it would only claim redpackets of $0.5 or more, but the airdrops of $0.001 will still be collected. This is because redpackets are the fastest way to get banned from a server - for example people doing redpackets for no value currencies. So this feature would allow you to decide for how much you would want to "sacrifice" your account (being banned). I buy 2021 discord accounts for $0.5 so I'd set the setting to like $0.6, to ensure profit while also being able to get a new account in case I get banned. (I buy accounts to bypass a certain server verification)
2. For the new feature where the account says "Thank you" after claiming a redpacket/airdrop (idk, I didn't try the new version yet), maybe add a setting for the account to only say this on airdrop/redpacket value of over $x.xx (it would be suspicious if I said thank you 10 times in a minute because I got 1 no value token xD)
3. Pause on ping option: The script will be paused for X minutes after the account received a ping in the channel where it claimed a drop. This would ensure that the script doesn't do any more suspicious things and (hopefully) people forget about you before the script is unpaused. "Ping received: Cooldown 3 minutes" (if the tip is from tip.cc bot, then no pause. Although this might be hard, since if someone tips you, you still get pinged by a user)
4. Autotyper: Every X minutes/seconds, the account will type a specified message in a specified channel. There could be multiple messages that could be randomized, and this would be used for messages like "can someone tip me any ltc please" (I'd use this in a drops-and-begging channel). So like after 3 minutes the acc types "Pls ltc", after the next 3 mins it types "pls btc" etc. The user can set as many messages as they like
Thanks for reading and coding! I'll buy the premium version as soon as I get $10 from the script xD

**Comment by jamestaylor141** [comment]
Also maybe settings per server 😅like settings.json but be able to have settings for each server you want to snipe drops in
My case, example where it would help: I would use this setting in 2 servers. Server1 usually does minidrops of $0.001 almost never drops over $0.1. Server2 does drops of $0.5 and more, and only does $0.001 minidrops when they want to detect bots. So in server1, I want to claim $0.001 drops, while in server2, if I claim these drops, I get $0.001 and a ban from the server

**Comment by jamestaylor141** [comment]
Sorry for adding more: Triviadrop seting where you can decide how many questions it should get right in percentage (like 70%). This is so that you don't get flagged as a bot for answering 20/20 trivias right. There could be an additional setting if triviadrop is over $x.xx it should invalidate the setting I mentioned before and always get the question right

**Comment by Itsmyalt** [comment]
Hi for some features I think there should be a feature where if one of the mods are online stop the autocollect or if he sends you a ping make it so it auto replys to the mod.
